### PR TITLE
Added Contracts from the Getting started Frontier Guide

### PR DESCRIPTION
**Pull request by alexvandesande**
Contracts used in the guide for the frontier:
https://docs.google.com/document/d/1fupguDpg-pB-EfBE7yqPEAtYQzXOo8iTd8rzIVggCs0/edit#heading=h.4iwpe7feo3ud

### CODE REVIEW

File: getting started/crowdsale.sol [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        amountRaised += f.amount;
+        tokenReward.sendToken(msg.sender, f.amount/price);
+        
+        return "thanks for your contribution";
+    }
+        
+    /* checks if the goal or time limit has been reached and ends the campaign */
+    function checkGoalReached() returns (bytes32 response) {
+        if (amountRaised >= fundingGoal){
+            uint i = 0;
```
Comment (chriseth) [inline getting started/crowdsale.sol]: Indentation looks strange here.

File: getting started/crowdsale.sol [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        return "thanks for your contribution";
+    }
+        
+    /* checks if the goal or time limit has been reached and ends the campaign */
+    function checkGoalReached() returns (bytes32 response) {
+        if (amountRaised >= fundingGoal){
+            uint i = 0; 
+            beneficiary.send(amountRaised);
+         suicide(beneficiary);
+         return "Goal Reached!";
```
Comment (chriseth) [inline getting started/crowdsale.sol]: Code after suicide is not executed.

File: getting started/crowdsale.sol [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        if (amountRaised >= fundingGoal){
+            uint i = 0; 
+            beneficiary.send(amountRaised);
+         suicide(beneficiary);
+         return "Goal Reached!"; 
+        }
+        else if (deadline <= block.number){
+            uint j = 0;
+            uint n = funders.length;
+            while (j <= n){
```
Comment (chriseth) [inline getting started/crowdsale.sol]: for-loop?

File: getting started/crowdsale.sol [hunk 4]
```diff
@@ -0,0 +1,76 @@
+contract token { mapping (address => uint) public balance; function token() {}  function sendToken(address receiver, uint amount) returns(bool sufficient) {  } }
```
Comment (chriseth) [inline getting started/crowdsale.sol]: You could use `function token();` here to make it explicit that it is only an interface.

File: getting started/token.sol [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        return true;
+    }
+
+    /* Try it yourself: merged mining
+
+    mapping (uint => address) miningReward;
+    function claimMiningReward() {
+        if (miningReward[block.number] == 0) {
+            balances[block.coinbase] += 1;
+    miningReward[block.number] = block.coinbase;
```
Comment (chriseth) [inline getting started/token.sol]: Indentation seems wrong.

File: getting started/crowdsale.sol (hunk 4 above)
Comment (alexvandesande) [inline getting started/crowdsale.sol]: I'm not sure I understand this, what exactly you meant?

File: getting started/crowdsale.sol (hunk 1 above)
Comment (alexvandesande) [inline getting started/crowdsale.sol]: Which line?

File: getting started/crowdsale.sol (hunk 2 above)
Comment (alexvandesande) [inline getting started/crowdsale.sol]: fixed

File: getting started/crowdsale.sol (hunk 3 above)
Comment (alexvandesande) [inline getting started/crowdsale.sol]: changed

File: getting started/token.sol (hunk 5 above)
Comment (alexvandesande) [inline getting started/token.sol]: fixed

File: getting started/crowdsale.sol (hunk 4 above)
Comment (chriseth) [inline getting started/crowdsale.sol]:
```
contract token {
 mapping (address => uint) public balance;
 function token();
 function sendToken(address receiver, uint amount) returns(bool sufficient);
}
```

Comment (alexvandesande) [inline getting started/crowdsale.sol]: I expanded the content into different lines, as requested.

On Tue, Jun 2, 2015 at 11:42 AM, chriseth <notifications@github.com> wrote:

> In getting started/crowdsale.sol
> <https://github.com/ethereum/dapp-bin/pull/22#discussion_r31529785>:
>
> > @@ -0,0 +1,76 @@
> > +contract token { mapping (address => uint) public balance; function token() {}  function sendToken(address receiver, uint amount) returns(bool sufficient) {  } }
>
> contract token {
>  mapping (address => uint) public balance;
>  function token();
>  function sendToken(address receiver, uint amount) returns(bool sufficient);
> }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ethereum/dapp-bin/pull/22/files#r31529785>.
>



-- 

Alex Van de Sande
UX Designer

**Pull request merged**